src/group_theory/p_group.lean

```diff
@@ -91,10 +91,24 @@ hG.of_surjective (quotient_group.mk' H) quotient.surjective_quotient_mk'
 lemma of_equiv {H : Type*} [group H] (ϕ : G ≃* H) : is_p_group p H :=
 hG.of_surjective ϕ.to_monoid_hom ϕ.surjective
 
+lemma pow_bijective' {n : ℕ} (hn : p.coprime n) : function.bijective ((^ n) : G → G) :=
```

Comment:
Can you make it an `equiv` like [pow_coprime](https://leanprover-community.github.io/mathlib_docs/group_theory/order_of_element.html#pow_coprime) with inverse `g ^ ((order_of g).gcd_b n)`? I think you can extract `(x.gcd_b y) % x = x.gcd y` that can be used to golf `left_inv` and `right_inv` of both `equiv`s.

Reply:
What I wrote isn't correct but this should be good enough combined with [pow_inj_mod](https://leanprover-community.github.io/mathlib_docs/group_theory/order_of_element.html#pow_inj_mod) and [int.mod_mod_of_dvd](https://leanprover-community.github.io/mathlib_docs/data/int/basic.html#int.mod_mod_of_dvd) (for pow_coprime):
```
import data.int.gcd

example (x y : ℕ) : ((y : ℤ) * x.gcd_b y) % x = (x.gcd y) % x :=
begin
  have := x.gcd_eq_gcd_ab y,
  rw [add_comm, ← sub_eq_iff_eq_add] at this,
  rw [← this, int.sub_mod, int.mul_mod_right, sub_zero, int.mod_mod],
end
```

Reply:
Your way can be made to work, but it's more painful than you might think. I've pushed one possible proof (using #17201), but maybe you can see a better?

Reply:
Actually, I was really overcomplicating things. I can just use my old proof of bijectivity.